src/aws-codeartifact/domain.ts

```diff
@@ -0,0 +1,338 @@
+      'sts:GetServiceBearerToken',
+    );
+  }
+}
+
+/**
+ * Interface representing the attributes of a CodeArtifact domain.
+ */
+export interface DomainAttributes {
+  /**
+   * The ARN (Amazon Resource Name) of the CodeArtifact domain.
+   */
+  readonly domainArn: string;
+
+  /**
+   * The name of the CodeArtifact domain.
+   */
+  readonly domainName: string;
+
+  /**
+   * The AWS KMS encryption key associated with the domain, if any.
+   */
+  readonly encryptionKey?: IKey;
+
+  /**
+   * The AWS account ID that owns the domain.
+   */
+  readonly domainOwner: string;
+}
+
+/**
+ * Construction properties for `Domain`.
+ */
+export interface DomainProps {
+  /**
+   * The name of the Domain
+   */
+  readonly domainName: string;
+  /**
+   * The key used to encrypt the Domain
+   */
+  readonly encryptionKey?: IKey;
+}
+
+/**
+ * Deploys a CodeArtifact domain.
+ */
+export class Domain extends DomainBase implements IDomain, ITaggableV2 {
+  /**
+   * Creates a Domain object from existing domain attributes.
+   *
+   * @param scope The parent construct.
+   * @param id The construct id.
+   * @param attrs The attributes of the domain to import.
+   */
+  public static fromDomainAttributes(scope: Construct, id: string, attrs: DomainAttributes): IDomain {
+    class Import extends DomainBase {
+      public readonly domainArn = attrs.domainArn;
+      public readonly domainName = attrs.domainName;
+      public readonly encryptionKey = attrs.encryptionKey;
+      public readonly domainOwner = attrs.domainOwner;
+      protected readonly policy?: PolicyDocument | undefined = undefined;
+    }
+
+    return new Import(scope, id);
+  }
+
+  /**
+   * Creates an IDomain object from an existing CodeArtifact domain ARN.
+   *
+   * @param scope The parent construct.
+   * @param id The construct id.
+   * @param domainArn - The ARN (Amazon Resource Name) of the existing CodeArtifact domain.
+   */
+  public static fromDomainArn(scope: Construct, id: string, domainArn: string): IDomain {
+    const domainResourceArnParts = Stack.of(scope).splitArn(domainArn, ArnFormat.SLASH_RESOURCE_NAME);
+    if (
+      domainResourceArnParts.resource !== 'domain' ||
+      domainResourceArnParts.account === undefined ||
+      domainResourceArnParts.resourceName === undefined
+    ) {
+      throw new Error(`Expected a domain ARN, but got ${domainArn}`);
+    }
+    return Domain.fromDomainAttributes(scope, id, {
+      domainArn,
+      domainName: domainResourceArnParts.resourceName,
+      domainOwner: domainResourceArnParts.account,
+    });
+  }
+
+  /**
+   * (internal) The CloudFormation resource representing this CodeArtifact domain.
+   */
+  protected cfnResource: CfnDomain;
+  /**
+   * The properties used to create the CloudFormation resource for this domain.
+   */
+  private cfnResourceProps: CfnDomainProps;
+
+  /**
+   * TagManager to set, remove and format tags
+   */
+  readonly cdkTagManager: TagManager;
```

Comment:
still needed?

Reply:
Yes, to implement the `ITaggableV2` interface. Or maybe I am still missing something

Reply:
@go-to-k do you know this?

Reply:
I think it is not necessary to implement this interface. This interface is for constructs with custom resources or `CfnResource` (without L1 resources), and all L1 constructs already implement the interface.

The following links are examples using V1 (`ITaggable`), which is almost identical to V2:

- ref: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/core/lib/tag-manager.ts#L277-L280
- example 1: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-eks/lib/fargate-profile.ts#L114
- example 2: https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-logs/lib/log-retention.ts#L140

@bweigel 

Are there any specific use cases for this?

Reply:
> Are there any specific use cases for this?

Not sure. Tbh I just thought V2 > V1 and I should implement this. Didn't think this is entirely redundant.

Reply:
There are no classes that implement those interfaces in L2Construct other than custom resources, so let's turn them off.